### PR TITLE
configs,ext,stdlib: Update DRAMSys integration

### PR DESCRIPTION
**Pull request by derchr**
Recent breaking changes in the DRAMSys API require user code to be
updated. These updates have been applied to the gem5 integration.

Furthermore, as DRAMSys started to use CMake dependency management,
it is no longer sensible to maintain two separate build systems for
DRAMSys. The use of the DRAMSys integration in gem5 will therefore
from now on require that CMake is installed on the target machine.

Additionally, support for snapshots have been implemented into DRAMSys
and coupled with gem5's checkpointing API.

Open TODOs:
- [ ] Currently the CMAKE_BUILD_TYPE is hardcoded to Release. Ideally it should be changed to Debug or RelWithDebInfo depending on the set scons build type. Is there a way to get the current build type from Python?

### CODE REVIEW

File: ext/dramsys/README [hunk 1]
```diff
@@ -1,11 +1,13 @@
-Follow these steps to get DRAMSys as part of gem5
+Follow these steps to build DRAMSys as part of gem5
 
 1. Go to ext/dramsys (this directory)
 2. Clone DRAMSys: 'git clone https://github.com/tukl-msd/DRAMSys DRAMSys'
-3. Change directory to DRAMSys: 'cd DRAMSys'
-4. Checkout the correct commit: 'git checkout -b gem5 09f6dcbb91351e6ee7cadfc7bc8b29d97625db8f'
-5. Recursively obtain the submodules: 'git submodule update --init --recursive'
+
+The latest verified working version is 8e021eada2cb937d5be24dbd02f3980e2193d6ec.
```
Comment (powerjg) [inline ext/dramsys/README]: Any chance you can point to a tag or stable release?

**Review by powerjg — COMMENTED** [review]
Overall this looks great. Just one small question.

PS: I think it's good to use cmake this way.

File: ext/dramsys/README (hunk 1 above)
Comment (derchr) [reply ext/dramsys/README]: Hi Jason, thanks for the quick review.
We will add a tag to make things easier for users.

Have you any ideas about the build type setting in scons?

Comment (powerjg) [reply ext/dramsys/README]: No, I have no idea how to get build information. gem5's scons build is crazy magic to me.

I guess what you're kinda going for would be "gem5.fast" would be like release and "gem5.opt" or "gem5.debug" would be "RelWithDebInfo" or "Debug", maybe? gem5 doesn't really have a "release" vs "debug" build. We just have a build with/without optimizations and with/without debug symbols.

Comment (derchr) [reply ext/dramsys/README]: Updated the readme to point to a tag.

Sadly, setting the correct CMake build type seems impossible as the external modules are included before the gem5 debug/opt/fast logic is run.

Comment (powerjg) [reply ext/dramsys/README]: No worries. I don't think it's terribly important to get it perfect.

**Comment by Harshil2107** [comment]
Hi @derchr,

All pull requests to gem5 require Change IDs to be present in all commit messages. The commits in this PR do not have Change IDs. Please follow these steps to add a change ID to your commits:

1. Run the following commands:
```
f=.git/hooks/commit-msg
mkdir -p 
curl -Lo $f https://gerrit-review.googlesource.com/tools/hooks/commit-msg
chmod +x $f/commit-msg
```

2. Then, amend the commit with `git commit --amend --no-edit`, and update your pull request.

**Comment by derchr** [comment]
> Hi @derchr,
> 
> All pull requests to gem5 require Change IDs to be present in all commit messages. The commits in this PR do not have Change IDs. Please follow these steps to add a change ID to your commits:
> 
>     1. Run the following commands:
> 
> 
> ```
> f=.git/hooks/commit-msg
> mkdir -p 
> curl -Lo $f https://gerrit-review.googlesource.com/tools/hooks/commit-msg
> chmod +x $f/commit-msg
> ```
> 
>     2. Then, amend the commit with `git commit --amend --no-edit`, and update your pull request.

Hi @Harshil2107 thanks for the guide. I updated the commit to include a change ID.

**Review by powerjg — APPROVED** [review]
Thanks for the updates!

**Pull request merged**